packages/constants/src/lib/utils/utils.ts

```diff
@@ -28,3 +30,26 @@ export function ERight<T>(result: T): IEither<T> {
     result,
   };
 }
+
+export async function getLatestEthBlockhash(): Promise<string> {
+  const provider = new ethers.providers.JsonRpcProvider(
+    LIT_EVM_CHAINS['ethereum'].rpcUrls[1]
+  );
+
+  for (let i = 1; i < LIT_EVM_CHAINS['ethereum'].rpcUrls.length; i++) {
+    try {
+      console.log('Fetching the latest Eth blockhash from RPC- ', i);
+      const latestEthBlockhash = (await provider.getBlock("latest")).hash;
+      console.log('latestEthBlockhash- ', latestEthBlockhash);
+
+      return latestEthBlockhash;
+    } catch (e: any) {
+      console.error(
+        'Error getting the latest Eth blockhash- ' + i + ', trying again: ',
+        e
+      );
+    }
+  }
+
+  throw new Error('Unable to get the latestBlockhash from any RPCs');
+}
```

Comment:
1. initialise provider inside loop to ensure each rpc url is used when attempting to fetch the latest block
2. start loop from index 0

Reply:
The first RPC isn't working properly & always returns the same nonce

Reply:
Fixed the provider new init. Also written a comment on the Ethereum RPCs

Reply:
re: The first RPC isn't working properly & always returns the same nonce

Can you change the RPC value in index 0 so that it won't return the same nonce. we need to start the loop index from 0

Reply:
also can we move the `getLatestEthBlockhash` function to `auth-browser`. It's not a constant so it shouldn't be in the `constants` package